src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,336 @@
+        }
+    }
+
+    // It is not possible to connect a cluster with its actual name, it seems a bug of the cloud API, that method is a workaround.
+    // There could be a better way to get this name, TODO: contact with cloud team.
+    private String getConnectionName(String clusterId) throws CloudException {
+        if (baseUrl.contains("uat.hazelcast.cloud"))
+            return String.format("ua-%s", clusterId);
+        else if (baseUrl.contains("cloud.hazelcast.com"))
+            return String.format("pr-%s", clusterId);
+        else {
+            throw new CloudException("Given base url isn't known and connection name cannot be created");
+        }
+    }
+
+    // It replaces the  value of token with *** in a response body. Such as bearer token, discovery token.
+    private String maskValueOfToken(String json)
+    {
+        return json.replaceAll("\"token\":\"(.*?)\"", "\"token\":\"******\"");
+    }
+
+    // It creates a folder with name clusterId under /home/user/
+    // Then it downloads the certificates for the cluster and put them in the created folder
+    private String downloadCertificatesAndGetPath(String clusterId) throws CloudException {
+        try
+        {
+            String userHome = System.getProperty("user.home");
+            Path pathClusterId = Paths.get(userHome, clusterId);
+
+            // If folder with clusterId is there then certificates are already downloaded
+            if(!Files.exists(pathClusterId))
+            {
+                createFolderAndDownloadCertificates(pathClusterId, clusterId);
+            }
+            return Paths.get(pathClusterId.toString(), "certificates").toString() + File.separator;
+        }
+        catch(Exception e)
+        {
+            throw new CloudException(String.format("Problem occurred during certificates download for cluster with id %s, Rc stack trace is: %s", clusterId, e.getStackTrace()));
+        }
+    }
+
+    private void createFolderAndDownloadCertificates(Path pathClusterId, String clusterId) throws IOException {
+        Path destination;
+        Files.createDirectories(pathClusterId);
+        Path pathResponseZip = Paths.get(pathClusterId.toString(), "certificates.zip");
+
+        destination = Paths.get(pathClusterId.toString(), "certificates");
+        Files.createDirectories(destination);
+
+        String cert_url = String.format("%s/cluster/%s/certificate", baseUrl, clusterId);
+        Request request = new Request.Builder()
+                .url(cert_url)
+                .get()
+                .header("Accept", "application/zip")
+                .header("Authorization", String.format("Bearer %s", bearerToken))
+                .build();
+
+        call = client.newCall(request);
+        Response response = call.execute();
+        try (FileOutputStream stream = new FileOutputStream(pathResponseZip.toString())) {
+            stream.write(response.body().bytes());
+        }
```

Comment:
try statement without a catch?

Reply:
The FileOutputStream in try block was used without catch statement generally, I don't know the exact reason.
But for our case, the `createFolderAndDownloadCertificates` method is called from `downloadCertificatesAndGetPath` and it has the catch statement, whenever there is an exception we will catch it and throw it.
But if you think it will be beneficial to add a catch statement there we can add it.

Reply:
```suggestion
        FileOutputStream stream = new FileOutputStream(pathResponseZip.toString());
        try {
            stream.write(response.body().bytes());
        } finally {
            stream.close();
        }        
```